### PR TITLE
Gift Entry: Batch Mode Edit Gift

### PR DESCRIPTION
**Pull request by kenlewis-sfdo**
This PR implements edit functionality when in Batch mode.  Gifts that have been entered into the table can be loaded back into the form using the "Open" row action.  Gifts that have not yet been processed will be editable in the form.  Gifts that have been processed/imported will load back into the form but will be prevented from being edited further by disabling the "Update" button.  

The "Cancel" button will reset the form, clearing fields or loading their default value if specified in the Template.

# Critical Changes

# Changes

# Issues Closed

# New Metadata
geActionOpen (label)
labelGeUpdate (label)

# Deleted Metadata

### CODE REVIEW

**Comment by kenlewis-sfdo** [comment]
**lurch: attach W-038208

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HpwcQAC" target="_blank">W-038208</a>

**Comment by kenlewis-sfdo** [comment]
**lurch: attach W-037858

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HpNRQA0" target="_blank">W-037858</a>

File: src/classes/BGE_BatchGiftEntry_UTIL.cls [hunk 1]
```diff
@@ -51,58 +51,89 @@ public with sharing class BGE_BatchGiftEntry_UTIL {
...
+            // to reference the fieldMappingByDevName property.
+            Map<String,BDI_FieldMapping> fieldMappingByDevName =
+                    BDI_MappingServiceAdvanced.getInstance().fieldMappingByDevName;
+
+            FORM_Template t = (FORM_Template) JSON.deserialize(
+                    batch.Form_Template__r.Template_JSON__c,
+                    FORM_Template.class
+            );
+
+            for (FORM_Section formSection : t.layout.sections) {
```
Comment (kenlewis-sfdo) [inline src/classes/BGE_BatchGiftEntry_UTIL.cls]: FYI @ptewson-sfdo this is the section where I swapped out collecting all mapped fields for getting all fields (and Name fields from related records) that are used in the Template.

File: src/classes/BGE_BatchGiftEntry_UTIL.cls [hunk 2]
```diff
@@ -51,58 +51,89 @@ public with sharing class BGE_BatchGiftEntry_UTIL {
...
+    public static Map<SObjectField, BDI_TargetFields> targetFieldsBySourceField = 
+        BDI_DataImportService.getDefaultMappingService().getTargetFieldsBySourceField();
+
+    /*******************************************************************************************************
+    * @description returns a list of DataImport__c fields the Batch Gift Entry UI needs in SOQL
+    * @param batchId the batch for which to get all fields for soql
+    * @param includeRelationshipFields whether to include relationship fields in the returned list
+    * @return List<String> list of DataImport__c field api names
+    */
+    public static List<String> getDataImportFields(Id batchId, Boolean includeRelationshipFields) {
```
Comment (ptewson-sfdo) [inline src/classes/BGE_BatchGiftEntry_UTIL.cls]: Can we move this second implementation of "getDataImportFields" up directly below the first one? Or move the first one down next to here?

Comment (ptewson-sfdo) [inline src/classes/BGE_BatchGiftEntry_UTIL.cls]: Can we move this second implementation of "getDataImportFields" up directly below the first one? Or move the first one down next to here?

File: src/classes/BGE_BatchGiftEntry_UTIL.cls [hunk 3]
```diff
@@ -51,58 +51,89 @@ public with sharing class BGE_BatchGiftEntry_UTIL {
...
+        List<String> fields = new List<String>();
```
Comment (ptewson-sfdo) [inline src/classes/BGE_BatchGiftEntry_UTIL.cls]: Maybe a more descriptive name for this ... fieldAPINames? (Are they API names?)

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 4]
```diff
@@ -18,12 +27,12 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
     @track version = '';
     @api showSpinner = false;
     @api batchId;
-    @api submissions = [];
     @api hasPageLevelError = false;
     label = { messageLoading, geSave, geCancel };
     @track formTemplateId;
     erroredFields = [];
     @api pageLevelErrorMessageList = [];
+    @track _data; // Row being updated when in update mode
```
Comment (ptewson-sfdo) [inline src/lwc/geFormRenderer/geFormRenderer.js]: Is there a less generic way to name this than "_data"?

File: src/classes/BGE_BatchGiftEntry_UTIL.cls (hunk 2 above)
Comment (kenlewis-sfdo) [reply src/classes/BGE_BatchGiftEntry_UTIL.cls]: Good call.  On it!

File: src/classes/BGE_BatchGiftEntry_UTIL.cls (hunk 3 above)
Comment (kenlewis-sfdo) [reply src/classes/BGE_BatchGiftEntry_UTIL.cls]: 👍🏽

File: src/lwc/geFormRenderer/geFormRenderer.js (hunk 4 above)
Comment (kenlewis-sfdo) [reply src/lwc/geFormRenderer/geFormRenderer.js]: Went with _dataRow - thanks for helping me hash that out!

**Comment by ptewson-sfdo** [comment]
@melissabarber I'm merging this in, but if you have docs related feedback we can follow up with an additional PR!

**Comment by melissabarber** [comment]
Thank you for the heads up @ptewson-sfdo. I've added rows for the two custom labels in this PR to be reviewed, and we'll clean them up with the final sweep.

**Pull request merged**